### PR TITLE
React save template

### PR DESCRIPTION
**Pull request by bjfletcher**
@ironsidevsquincy There's the existing HTML template which I've ported to become a React component. I've gone with a different boilerplate where properties are defined in the arguments rather than in comments. I hope this is OK? I prefer this approach.

### CODE REVIEW

**Comment by ironsidevsquincy** [comment]
It would be good to have a comment above to explain what the arguments mean/what type they are (e.g. `myFtApiWrite`?)

File: myft/templates/save-for-later.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+export default ({ myFtApiWrite, contentId, variant, buttonText }) => {
```
Comment (ironsidevsquincy) [inline myft/templates/save-for-later.js]: in general, arguments in an object are optional. looks like `contentId` is a required argument, so would right this as

```
(contentId, { myFtApiWrite, variant, buttonText })
```

File: myft/templates/save-for-later.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	return !myFtApiWrite ? null : (
```
Comment (ironsidevsquincy) [inline myft/templates/save-for-later.js]: isn't of negating, why not

```
myFtApiWrite ?
    <form ...>
     ...
   </form> : 
   null;
```

File: myft/templates/save-for-later.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+					type="submit"
+					className={`n-myft-ui__button${variant ? 'n-myft-ui__button--' + variant : ''}`}
+					data-trackable="save-for-later"
+					data-alternate-label="Saved to myFT"
+					data-alternate-text="Saved"
+					data-content-id={contentId}
+					aria-label="Save to myFT"
+					title="Save to myFT"
+			>{buttonText || 'Save'}</button>
+		</form>
```
Comment (ironsidevsquincy) [inline myft/templates/save-for-later.js]: is this indentation a bit off?

File: myft/templates/save-for-later.js (hunk 2 above)
Comment (bjfletcher) [inline myft/templates/save-for-later.js]: For readability. I think it's better to have the very large block at the end, rather in the middle; if you want to know what happens if it's false, you'd have to scroll all the way down to see that it's `null`. What do you think?

File: myft/templates/save-for-later.js (hunk 3 above)
Comment (bjfletcher) [inline myft/templates/save-for-later.js]: I tend to like have 2 tabs for continuing lines, 1 tab for actual indention. If we have 1 tab for continuing lines, then it's hard to distinguish between continuing lines and actual indention. Interested to know your opinion on this.

File: myft/templates/save-for-later.js (hunk 1 above)
Comment (bjfletcher) [inline myft/templates/save-for-later.js]: That's great. :+1: :)

File: myft/templates/save-for-later.js (hunk 3 above)
Comment (ironsidevsquincy) [inline myft/templates/save-for-later.js]: Looks funny to my eye, but don't have a strong opinion about it

File: myft/templates/save-for-later.js (hunk 2 above)
Comment (ironsidevsquincy) [inline myft/templates/save-for-later.js]: Always find negations harder to understand, but that's just me, don't have a string opinion about it

**Pull request merged**